marlowe-dashboard-client/src/Wallet/Types.purs

```diff
@@ -0,0 +1,41 @@
+module Wallet.Types
+  ( WalletLibrary
+  , WalletNicknameKey
+  , WalletDetails(..)
+  , PubKeyHash
+  ) where
+
+import Prelude
+import Data.Generic.Rep (class Generic)
+import Data.Map (Map)
+import Data.Newtype (class Newtype)
+import Data.Tuple (Tuple)
+import Foreign.Generic (class Decode, class Encode, defaultOptions, genericDecode, genericEncode)
+
+type WalletLibrary
+  = Map WalletNicknameKey WalletDetails
+
+-- N.B. make the nickname (string) the first key so we get alphabetical ordering for free
+type WalletNicknameKey
+  = Tuple String PubKeyHash
+
+newtype WalletDetails
```

Comment:
why a newtype? if it's for JSON, I don't think we need it because I think `encode { userHasPickedUp: true }` will just work?